### ISSUE TITLE
index.html does not respect __all__

### ISSUE THREAD

**Issue by fweisser** [mitmproxy/pdoc#221]
# Description
When I try to exclude certain submodules from a project via `__all__`, they don't appear in the "normal" view, but they appear in the index.html search page.

It seems that `pkgutil.walk_packages` (https://github.com/mitmproxy/pdoc/blob/1dfd11baff29ae92f0e2d1b3725e7c57129aa6dd/pdoc/extract.py#L83) does not respect `__all__`.

# Steps to reproduce
1. create the following project structure
    ```
    mod
    ├── __init__.py
    ├── submod1
    │   └── __init__.py
    └── submod2
        ├── __init__.py
        ├── subsubmod1
        │   └── __init__.py
        └── subsubmod2
            └── __init__.py
    ```
2. exclude `mod.submod2.subsubmod2` from documentation:
    *mod/submod2/\_\_init__.py*:
    ```python
    __all__ = [ "subsubmod1" ]
    ```
3. generate documentation of `mod`
4. navigate to `mod.submod2`: `mod.submod2.subsubmod2` is not shown
5. navigate to `index.html`: `mod.submod2.subsubmod2` appears

**Comment by mhils** [comment]
Thanks! `__all__` unfortunately isn't really specced anywhere, but after re-reading https://docs.python.org/3/tutorial/modules.html#importing-from-a-package I agree that if  `__all__` is specified, modules outside `__all__` should be ignored. I've fixed this in #223 and just pushed pdoc 6.3.0. 😃

**Comment by fweisser** [comment]
Great, thanks for the quick fix!
